src/jobs/deploy-node-project.yml

```diff
@@ -0,0 +1,60 @@
+description: |
+  Build, test and deploy a basic NodeJS project.
+  This job runs `npm install`, `npm run build` and `npm run test`,
+  and on success, runs `npm ci --production` and deploys the current project to the requested instance.
+
+executor: default
+
+parameters:
+
+  instance-name:
+    type: string
+    description: "The instance this version will be deployed to"
+
+  env-name:
+    type: string
+    description: "The environment containing this instance"
+
+  image-name:
+    type: string
+    description: "Optional. Image name to use when pushing this version"
+    default: ""
+
+steps:
+
+  - checkout
+
+  - restore_cache:
+      keys:
+      - v1-dependencies-
+      # fallback to using the latest cache if no exact match is found
+      - v1-dependencies-
+
+  - setup
+
+  - run:
+      name: Install dependecies
+      command: |
+        npm ci
+        npm run build
+
+  - save_cache:
```

Comment:
The cache here is not being checked against any file, which will lead to a stale cache. https://circleci.com/docs/2.0/caching/#example-caching-configuration

Consider syncing the cache to the checksum of the `package.lock` file. This will clear the cache when this file changes.